estuary/models/koji.py

```diff
@@ -28,6 +31,31 @@ class KojiBuild(EstuaryStructuredNode):
     tasks = RelationshipFrom('KojiTask', 'TRIGGERED')
     version = StringProperty()
 
+    @classmethod
+    def find_or_none(cls, identifier):
+        """
+        Find the node using the supplied identifier.
+
+        :param str identifier: the identifier to search the node by
+        :return: the node or None
+        :rtype: EstuaryStructuredNode or None
+        """
+        uid = identifier
+        if re.match(r'^\d+$', uid):
+            # The identifier is an ID
+            return cls.nodes.get_or_none(id_=uid)
+        elif uid.endswith('.src.rpm'):
+            # The identifer is likely an NVR with .src.rpm at the end, so strip that part of it
+            # so it can be treated like a normal NVR
+            uid = uid[:-8]
+
+        if len(uid.rsplit('-', 2)) == 3:
```

Comment:
Can we support this with another heuristic as well? This doesn't look bullet proof

Reply:
I thought maybe there was, but I took the applicable code from:
https://github.com/release-engineering/kobo/blob/master/kobo/rpmlib.py#L166

Reply:
cool, +1